### PR TITLE
Remove username test: Re-launch test for plan-specific flows

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

For context, please check pbxNRc-rq#comment-959

* The remove-username test that was launched in https://github.com/Automattic/wp-calypso/pull/45609 came out a winner for the control group in the `onboarding` flow, and we've stopped the test. This PR relaunches the test for the plan-specific flows - /personal, /premium, /business, /ecommerce
* Remove the A/A test that was launched in https://github.com/Automattic/wp-calypso/pull/45770

#### Testing instructions

* Begin signup at the plan-specific flows - http://calypso.localhost:3000/start/personal, http://calypso.localhost:3000/start/premium, http://calypso.localhost:3000/start/business, and http://calypso.localhost:3000/start/ecommerce. 
    * Verify that you are assigned to the test `removeUsernameInSignup`.
    * If in `control, you should see the username field; if in `variantShowUsername` then you should not see the username field.
    * Verify that you can successfully complete signup for both control and variant.
    * Check the username assigned to you - it should be based on your email address.
    * Verify that the account activation email's subject line mentions your full email address.
* Verify that you are _not assigned_ to the test if in the `onboarding` flow, or in Jetpack signup flow.
* Verify that the A/A test `usernameAATest` is removed.

### CODE REVIEW

File: client/blocks/signup-form/index.jsx [hunk 1]
```diff
@@ -1067,17 +1067,11 @@ function TrackRender( { children, eventName } ) {
...
-			'premium',
-			'business',
-			'ecommerce',
-		];
+		const eligibleFlowsForRemoveUsernameTest = [ 'personal', 'premium', 'business', 'ecommerce' ];
 		let displayUsernameInput = true;
 
 		if ( eligibleFlowsForRemoveUsernameTest.includes( ownProps.flowName ) ) {
-			abtest( 'usernameAATest' );
+			'control' === abtest( 'removeUsernameInSignup' );
```
Comment (aneeshd16) [inline client/blocks/signup-form/index.jsx:1074]:
```suggestion
			displayUsernameInput = 'control' === abtest( 'removeUsernameInSignup' );
```
You probably meant this? :)

**Review by aneeshd16 — CHANGES_REQUESTED** [review]

File: client/blocks/signup-form/index.jsx (hunk 1 above)
Comment (niranjan-uma-shankar) [reply client/blocks/signup-form/index.jsx]: Ah yes, I forgot to push that change, it was in my local commit. Just pushed, thanks for catching that.

Comment (niranjan-uma-shankar) [reply client/blocks/signup-form/index.jsx]: The commit [7ed2a28](https://github.com/Automattic/wp-calypso/pull/46038/commits/7ed2a287b3b901115aa3828b3754fece5c9a7bbd) also removes the A/A test.

**Review by aneeshd16 — APPROVED** [review]
Tested all cases, LGTM 👍

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@niranjan-uma-shankar please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/914238). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Pull request merged**